src/classes/ADDR_Addresses_TDTM.cls

```diff
@@ -67,231 +82,231 @@ public class ADDR_Addresses_TDTM extends TDTM_Runnable {
+		            if (changedToDefault ||seasonalChanged) {
+		                putAddrInMap(addr, mapParentIdAddr);
+		            //If it's default and its address info has changed.
+		            } else if (changed && addr.Default_Address__c) {
+		                putAddrInMap(addr, mapParentIdAddr);    
+		            }
+		            i++;
 	            }
 	            
-	            // AFTER UPDATE
-	            if (triggerAction == TDTM_Runnable.Action.AfterUpdate) {
-	                // an address that is marked default needs to propogate to the household
-	                if ((addr.Default_Address__c && (addr.Default_Address__c != addrOld.Default_Address__c)) 
-	                    // any seasonal address also need to be considered
-	                    || isSeasonalChanged(addr, addrOld)) {
-	                    mapAccIdAddr.put(addr.Parent_Account__c, addr);
-	                } else if (isAddressChanged(addr, addrOld, true)) {
-	                    if (addr.Default_Address__c)
-	                        mapAccIdAddr.put(addr.Parent_Account__c, addr);
-	                    else    // address changed, but not default, still need to update any contacts using it as an override
-	                        mapAddrIdAddr.put(addr.Id, addr);      
-	                } 
+	            if(mapParentIdAddr.size() > 0) {
+	                refreshCurrentAccAddress(mapParentIdAddr, dmlWrapper, triggerAction);
+	                processDml(dmlWrapper, mapParentIdAddr);
 	            }
 	            
-	            // moving to the next item
-	            i++;            
+	            alreadyRunAfterUpdate = true;
 	        }
-	
-	        // BEFORE INSERT & BEFORE UPDATE
-	        if (triggerAction == TDTM_Runnable.Action.BeforeInsert ||
-	           triggerAction == TDTM_Runnable.Action.BeforeUpdate) {
-	            // enforce address records created only as children of Accounts
-	            verifyAddrAccChildOnly((list<Address__c>)listNew);
+	        
+	        // AFTER DELETE
+	        if (!alreadyRunAfterDelete && triggerAction == TDTM_Runnable.Action.AfterDelete) {
+	            Integer i = 0;
+	            for (SObject addrOld : listOld) {
+		            // first go through all new/modified Addresses, and collect the list of parents to consider. 
+		            ID parentAccId = ((Address__c)addrOld).Parent_Account__c;
+		            ID parentConId = ((Address__c)addrOld).Parent_Contact__c;
+		            if (parentAccId != null) {
+		                mapParentIdAddr.put(parentAccId, null);
+		            } else if(parentConId != null) {
+		                mapParentIdAddr.put(parentConId, null);
+		            }
+	                i++;
+	            }
 	            
-	            // enforce Seasonal Addresses don't overlap
-	            verifyAddrSeasonalNoOverlap((list<Address__c>)listNew);
+	            if(mapParentIdAddr.size() > 0) {
+	               refreshCurrentAccAddress(mapParentIdAddr, dmlWrapper, triggerAction);
```

Comment:
i was confused by this method name, since it says AccAddress, but your parent map might have contacts as well.  I see your method supports both, but you should consider updating the name to make this clearer.